module/VuFind/src/VuFind/View/Helper/Root/RecordDataFormatter.php

```diff
@@ -131,20 +153,35 @@ protected function render($driver, $field, $data, $options)
 
     /**
      * Create formatted key/value data based on a record driver and field spec.
+     * The first argument can be a descendant of RecordDriver.
+     * If omitted, then invoke this class with the desired driver.
+     * The second or first argument is an array containing formatting specifications.
      *
-     * @param RecordDriver $driver Record driver object.
-     * @param array        $spec   Formatting specification
+     * @param array ...$args Record driver object and/or formatting specifications.
      *
      * @return array
      */
-    public function getData(RecordDriver $driver, array $spec)
+    public function getData(...$args)
     {
+        if (empty($args[0])) {
+            return [];
+        }
+        if ($args[0] instanceof RecordDriver) {
```

Comment:
Should we do the instanceof check BEFORE the empty check? Otherwise, if you invoke the method with a driver and an empty array, we'll end up doing extra work.

Reply:
I could do that! Yeah, I can alter the templates to use the new invocation.